cmd/task/task.go

```diff
@@ -28,19 +28,34 @@ func main() {
 			Color:   flags.Color,
 		}
 		if err, ok := err.(*errors.TaskRunError); ok && flags.ExitCode {
+			emitCIErrorAnnotation(err)
 			l.Errf(logger.Red, "%v\n", err)
 			os.Exit(err.TaskExitCode())
 		}
 		if err, ok := err.(errors.TaskError); ok {
+			emitCIErrorAnnotation(err)
 			l.Errf(logger.Red, "%v\n", err)
 			os.Exit(err.Code())
 		}
+		emitCIErrorAnnotation(err)
 		l.Errf(logger.Red, "%v\n", err)
 		os.Exit(errors.CodeUnknown)
 	}
 	os.Exit(errors.CodeOk)
 }
 
+// emitCIErrorAnnotation emits an error annotation for supported CI providers.
+func emitCIErrorAnnotation(err error) {
+	if os.Getenv("GITHUB_ACTIONS") != "true" {
```

Comment:
[`ParseBool`](https://pkg.go.dev/strconv#ParseBool) is usually a good idea in these scenarios. GA might be fixed at `true`, but still...

```suggestion
	if isGA, _ := strconv.ParsePool(os.Getenv("GITHUB_ACTIONS")); !isGA {
```